src/core/types/decimal192.rs

```diff
@@ -207,38 +413,118 @@ impl Decimal {
+            ///
+            /// assert!(SUT::zero().is_zero());
+            /// assert!(!SUT::one().is_zero());
+            /// assert!(!SUT::one().neg().is_zero());
+            /// ```
+            ///
             pub fn is_zero(&self) -> bool;
 
             /// Whether this decimal is positive.
+            ///
+            /// ```
+            /// extern crate sargon;
+            /// use sargon::prelude::*;
+            /// #[allow(clippy::upper_case_acronyms)]
+            /// type SUT = Decimal;
+            ///
+            /// assert!(SUT::one().is_positive());
+            /// assert!(!SUT::zero().is_positive());
+            /// assert!(!SUT::one().neg().is_positive());
+            /// ```
+            ///
             pub fn is_positive(&self) -> bool;
 
             /// Whether this decimal is negative.
+            ///
+            /// ```
+            /// extern crate sargon;
+            /// use sargon::prelude::*;
+            /// #[allow(clippy::upper_case_acronyms)]
+            /// type SUT = Decimal;
+            ///
+            /// assert!(SUT::one().neg().is_negative());
+            /// assert!(!SUT::one().is_negative());
+            /// assert!(!SUT::zero().is_negative());
+            /// ```
+            ///
             pub fn is_negative(&self) -> bool;
         }
     }
 }
 
 impl Decimal {
-    pub fn checked_powi(&self, exp: i64) -> Option<Self> {
+    /// Creates the Decimal `10^exponent`, returns `None` if overflows.
+    #[inline]
+    pub(crate) fn checked_powi(&self, exp: i64) -> Option<Self> {
         self.native().checked_powi(exp).map(|n| n.into())
     }
 
     /// Creates the Decimal `10^exponent`
+    ///
+    /// ```
+    /// extern crate sargon;
+    /// use sargon::prelude::*;
+    /// #[allow(clippy::upper_case_acronyms)]
+    /// type SUT = Decimal192;
+    ///
+    /// assert_eq!(SUT::pow(2).to_string(), "100");
+    /// assert_eq!(SUT::pow(3).to_string(), "1000");
+    /// ```
+    ///
+    #[inline]
     pub fn pow(exponent: u8) -> Self {
         Self::from(10)
             .checked_powi(exponent as i64)
             .expect("Too large exponent, 10^39 is max.")
     }
 
-    /// `abs(self)`
-    /// Panics if Self is Self::MIN.
+    /// Returns the absolute value, if `self` is `Decimal::min()` then `Decimal::max()`
+    /// is returned, since `Decimal::min().abs()` would overflow.
+    ///
+    /// ```
+    /// extern crate sargon;
+    /// use sargon::prelude::*;
+    /// #[allow(clippy::upper_case_acronyms)]
+    /// type SUT = Decimal192;
+    ///
+    /// assert_eq!(SUT::two().neg().abs(), SUT::two());
+    /// assert_eq!(SUT::three().abs(), SUT::three());
+    /// assert_eq!(SUT::max().abs(), SUT::max());
+    /// ```
+    ///
+    #[inline]
     pub fn abs(&self) -> Self {
-        self.native().checked_abs().expect("Expected clients of Sargon to not use so large negative numbers (Self::MIN).").into()
+        if self == &Self::min() {
+            Self::max()
```

Comment:
Is this reasonable to do?! I think it is! We now return `Decimal::max()` instead of CRASH if we call `abs` on `Decimal::min()`, they are aaaaaaalmost eq. We also note that it is probably never going to happen that our Wallets use `Decimal::min()`, we almost never use negative numbers at all...